src/python/dxpy/scripts/dx.py

```diff
@@ -427,27 +414,24 @@ def logout(args):
         if state["interactive"]:
             dxpy.AUTH_HELPER = None
         else:
-            write_env_var("DX_SECURITY_CONTEXT", None)
+            dxpy.config.write("DX_SECURITY_CONTEXT", None)
 
 def set_api(protocol, host, port, write):
-    os.environ['DX_APISERVER_PROTOCOL'] = protocol
-    os.environ['DX_APISERVER_HOST'] = host
-    os.environ['DX_APISERVER_PORT'] = port
+    dxpy.config.update(DX_APISERVER_PROTOCOL=protocol,
+                       DX_APISERVER_HOST=host,
+                       DX_APISERVER_PORT=port)
     if write:
-        write_env_var("DX_APISERVER_PROTOCOL", protocol)
-        write_env_var("DX_APISERVER_HOST", host)
-        write_env_var("DX_APISERVER_PORT", port)
-    dxpy.set_api_server_info(host=host, port=port, protocol=protocol)
+        dxpy.config.save()
 
 def set_project(project, write, name=None):
     if dxpy.JOB_ID is None:
         os.environ['DX_PROJECT_CONTEXT_ID'] = project
         if name is not None:
-            set_env_var("DX_PROJECT_CONTEXT_NAME", name)
+            dxpy.config["DX_PROJECT_CONTEXT_NAME"] = name
         if write:
-            write_env_var("DX_PROJECT_CONTEXT_ID", project)
+            dxpy.config.write("DX_PROJECT_CONTEXT_ID", project)
             if name is not None:
-                write_env_var("DX_PROJECT_CONTEXT_NAME", name)
+                dxpy.config.write("DX_PROJECT_CONTEXT_NAME", name)
```

Comment:
Previously these write operations were mostly writing different files, but now they write all the files (resulting in each config file being written multiple times). Consider batching the writes with a single "save" call at the end.